### ISSUE TITLE
Documentation on interval

### ISSUE THREAD

**Issue by alexkwolfe** [jimeh/redistat#15]
I noticed that there is both an `interval` and a `depth` option for the finder.  What's unclear to me is how those options are meant to be used.

The goal is to be able to find stats and return them aggregated as one number or return them broken out into multiple numbers depending on the size of the time interval used (hourly, daily, yearly). What's the best way to accomplish that?

Intuitively, I'd think that `interval` would specify the size of the interval returned by the finder.  This seems to be the case:

``` ruby
require 'redistat'

class ViewStats
  include Redistat::Model
  depth :hour
end

ViewStats.store('hello', {:world => 4}, 1.hour.ago)
ViewStats.store('hello', {:world => 2}, 3.hours.ago)

ViewStats.find('hello', 1.hour.ago, 1.hour.from_now, :interval => :hour).all
 => [{"world"=>"4"}, {}, {}]
```

And I'd think that omitting interval would cause the finder to return only one interval.  This also seems to be the case:

``` ruby
ViewStats.find('hello', 3.hours.ago, 1.hour.from_now).all
 => [{"world"=>6}]
```

I would expect that using a `year` interval would also cause the finder to return only one interval (assuming there had been stats collected in only one year).  But this is not the case. The finder behaves as though I used the hour interval:

``` ruby
ViewStats.find('hello', 1.hour.ago, 1.hour.from_now, :interval => :year).all
 => [{"world"=>"4"}, {}, {}] 
```

I see that the default `depth` option is `:hour`, so I thought maybe this was interfering with interval working as expected. Setting `depth` to `:year` causes the time range to be ignored:

``` ruby
ViewStats.find('hello', 1.hour.ago, 1.hour.from_now, :interval => :year, :depth => :year).all
 => [{"world"=>"6"}]
```

So it's unclear to me how these options are meant to be used.  Any additional clarity you can provide would be immensely appreciated.

**Comment by alexkwolfe** [comment]
It occurs to me that there's a bug here, but I could be wrong.  

I guess the last example I gave is expected, since I set the depth to :year. The finder is only querying for 2012: `hgetall ViewStats/hello:2012`. 

What's unexpected is that `:interval => :year, :depth => :hour` would return more than one result.  At least I think it's unexpected.  Do you agree?

**Comment by jimeh** [comment]
I agree, that does look like a bug. I'll try to dig into it tomorrow :)
